### DISCUSSION TITLE
Radius for `topaz train`

### DISCUSSION THREAD

**Discussion by annwhoorma** [tbepler/topaz#190] [General]
`train` command accepts radius as an optional argument, and in most places it's set to either 1, 2, or 3 (default), regardless of what actual radius in _pix_ is. For example, this is the radius description in `train` script:

```python
training.add_argument('-r', '--radius', default=3, type=int, help='pixel radius around particle centers to consider positive (default: 3)')
```
However, the actual positive area particle center is usually larger (8 and more, even with downscaling). Even `extract` command reflects radius boundaries (for tuning) between 5 and 100.

When I tried to increase the radius for training, I got worse results than with a smaller radius. Why the radius for training don't seem to correspond to an actual radius of particles?

**Comment by tbepler** [comment]
The radius for `train` and the radius for `extract` do not relate to the same underlying concepts. The `train` radius specifies the set of pixels around the center coordinate of each labeled particle that are included as positive labels for training the classifier. Setting this to a small number (much smaller than the radius of the particle) is important for getting well centered predictions. The `extract` radius specifies how close predicted particles can be to each other when coordinates are identified from the pixel-level probabilities via peak finding. This radius should be close to the actual size of your particle, because it prevents multiple high scoring pixels within a single particle being called as centers of different particles by the peak finding algorithm.

**Reply by annwhoorma** [reply]
Understood, thanks a lot for giving a thorough explanation! 

it'd be nice if the code or documentation had this explanation to avoid confusion :)

**Reply by tbepler** [reply]
There's some description of these parameters in this [walkthrough](https://github.com/tbepler/topaz/blob/master/tutorial/02_walkthrough.ipynb), but I agree we should improve this documentation. Thanks for the feedback!
